csharp/src/Ice/Communicator.cs

```diff
@@ -492,16 +477,20 @@ public Properties getProperties()
         /// <returns>This communicator's logger.
         ///
         /// </returns>
-        public Logger getLogger()
+        public Logger Logger
         {
-            return _initData.logger;
+            get
+            {
+                Debug.Assert(_initData.logger != null);
```

Comment:
Do we need these Debug.Assert?

Reply:
That was to prevent nullable warning, because _initData.logger was nullable field, I'm removing the InitializationData now with next round all changes and this will be gone.